[Cutlass] Import cutlass python API for EVT

mlazos opened the pull request:
This imports the pieces of the cutlass python API that are needed for python EVT tracing. It builds on existing importing for cutlass_library. Once EVT tracing has been added to cutlass_library (should be later this year) this can be removed.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #150910
* #150909
* #150908
* #150907
* #150906
* #150905
* #150904
* #150903
* #150346
* #150345
* __->__ #150344



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

mlazos committed:
[Cutlass] Import cutlass python API for EVT

[ghstack-poisoned]

henrylhtsang commented:
can you insert a pdb.set_trace in the except clause, and then run 
```
python -x test/inductor/test_cutlass_backend.py
```
to see if that works?

henrylhtsang commented:
Also can you add a unit test to check for the import? This can help when we update the cutlass version

mlazos committed:
Update on "[Cutlass] Import cutlass python API for EVT"

This imports the pieces of the cutlass python API that are needed for python EVT tracing. It builds on existing importing for cutlass_library. Once EVT tracing has been added to cutlass_library (should be later this year) this can be removed.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
umm I checkout your commit, and test with 
```
python test/inductor/test_cutlass_backend.py -k test_import_cutlass 
```
and it didn't work.

mlazos committed:
Update on "[Cutlass] Import cutlass python API for EVT"

This imports the pieces of the cutlass python API that are needed for python EVT tracing. It builds on existing importing for cutlass_library. Once EVT tracing has been added to cutlass_library (should be later this year) this can be removed.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

mlazos committed:
Update on "[Cutlass] Import cutlass python API for EVT"

This imports the pieces of the cutlass python API that are needed for python EVT tracing. It builds on existing importing for cutlass_library. Once EVT tracing has been added to cutlass_library (should be later this year) this can be removed.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

mlazos committed:
Update on "[Cutlass] Import cutlass python API for EVT"

This imports the pieces of the cutlass python API that are needed for python EVT tracing. It builds on existing importing for cutlass_library. Once EVT tracing has been added to cutlass_library (should be later this year) this can be removed.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented on torch/_inductor/codegen/cuda/cutlass_lib_extensions/cutlass_mock_imports/cuda/__init__.py:
```diff
@@ -0,0 +1,4 @@
+__version__ = "12.6"
```
not a big fan of this but okay

mlazos replied:
I could set it to None, I forgot I left this lol

mlazos replied:
Called the torch API to get this now :)

henrylhtsang commented on torch/_inductor/codegen/cuda/cutlass_lib_extensions/cutlass_mock_imports/cuda/cuda.py:
```diff
@@ -0,0 +1,13 @@
+    pass
+
+class CUstream:
+    def __init__(self, v):
```
lint MYPY [no-untyped-def]

henrylhtsang commented on torch/_inductor/codegen/cuda/cutlass_utils.py:
```diff
@@ -75,28 +76,72 @@ def try_import_cutlass() -> bool:
+
         try:
+            breakpoint()
+            import cutlass  # noqa: F401
```
you might consider importing pycute etc as well, if you need it

henrylhtsang reviewed (approved):
it can pass if I remove the breakpoint.

mlazos committed:
Update on "[Cutlass] Import cutlass python API for EVT"

This imports the pieces of the cutlass python API that are needed for python EVT tracing. It builds on existing importing for cutlass_library. Once EVT tracing has been added to cutlass_library (should be later this year) this can be removed.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

mlazos committed:
Update on "[Cutlass] Import cutlass python API for EVT"

This imports the pieces of the cutlass python API that are needed for python EVT tracing. It builds on existing importing for cutlass_library. Once EVT tracing has been added to cutlass_library (should be later this year) this can be removed.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

mlazos committed:
Update on "[Cutlass] Import cutlass python API for EVT"

This imports the pieces of the cutlass python API that are needed for python EVT tracing. It builds on existing importing for cutlass_library. Once EVT tracing has been added to cutlass_library (should be later this year) this can be removed.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

mlazos committed:
Update on "[Cutlass] Import cutlass python API for EVT"

This imports the pieces of the cutlass python API that are needed for python EVT tracing. It builds on existing importing for cutlass_library. Once EVT tracing has been added to cutlass_library (should be later this year) this can be removed.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

mlazos commented:
@pytorchbot merge
